### PR TITLE
docs: fix broken line numbers

### PR DESCRIPTION
**Pull request by chenxsan**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Closes https://github.com/eslint/eslint/issues/16042.

1. Enable [`templateFormats`](https://github.com/11ty/eleventy-plugin-syntaxhighlight/blob/35987a840890fcf5dc8d90d370e653d088fccdf2/.eleventy.js#L27) for `@11ty/eleventy-plugin-syntaxhighlight` to stop it from processing code block for markdown
2. Add some `markdown-it` plugins to generate line numbers, which are based on [vitepress's implementation](https://github.com/vuejs/vitepress/tree/main/src/node/markdown/plugins)

<details><summary>Before</summary>

[https://eslint.org/docs/latest/rules/function-paren-newline#when-not-to-use-it](https://eslint.org/docs/latest/rules/function-paren-newline#when-not-to-use-it)

eslint org_docs_latest_rules_function-paren-newline

</details>

<details><summary>After</summary>

[https://deploy-preview-16606--docs-eslint.netlify.app/rules/function-paren-newline#when-not-to-use-it](https://deploy-preview-16606--docs-eslint.netlify.app/rules/function-paren-newline#when-not-to-use-it)

deploy-preview-16606--docs-eslint netlify app_rules_function-paren-newline (1)

</details>

And here's a video where I'm switching tabs between before and after:

https://user-images.githubusercontent.com/1091472/205433482-267cded1-9af6-4e4e-94f0-e38522eedd70.mov

#### Is there anything you'd like reviewers to focus on?

Note that this is more like a workaround because the upstream won't fix it in near future from what I can tell when 
[the feature request for showing line numbers](https://github.com/11ty/eleventy-plugin-syntaxhighlight/issues/10) has been proposed for almost 4 years now with two old pull requests without any response from the maintainer.

And I believe with this pull request we can just remove `@11ty/eleventy-plugin-syntaxhighlight`.

### CODE REVIEW

File: docs/src/assets/scss/syntax-highlighter.scss [hunk 1]
```diff
@@ -100,25 +100,32 @@ pre[class*="language-"] {
...
+}
+
+div[class*="language-"].line-numbers-mode {
+    position: relative;
+}
+
+div[class*="language-"].line-numbers-mode pre[class*="language-"] {
+    padding-left: calc(2.4em + 1.2rem);
+    margin-top: 0;
+    margin-bottom: 0;
```
Comment (harish-sethuraman) [inline docs/src/assets/scss/syntax-highlighter.scss:130]:
```suggestion
    width: 1.2em;
}

div[class*="language-"].line-numbers-mode {
    position: relative;
    pre[class*="language-"] {
        padding-left: calc(2.4em + 1.2rem);
        margin-top: 0;
        margin-bottom: 0;
        
    }
}
```
maybe?

**Review by kecrily — APPROVED** [review]
LGTM

**Review by harish-sethuraman — APPROVED** [review]
LGTM, Just make sure nothing has broken. This is good to merge 👍🏻

**Review by amareshsm — APPROVED** [review]
LGTM

File: docs/mdSyntaxHighlighter.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/** @typedef {import("markdown-it")} MarkdownIt */
```
Comment (nzakas) [inline docs/mdSyntaxHighlighter.js:1]: This isn't the right location for this file. Can you please move it to `src/_plugins` and rename to `md-syntax-highlighter.js`?

Also, did you copy any of this code from somewhere else? If so, we need to include the license information.

**Review by nzakas — CHANGES_REQUESTED** [review]

File: docs/mdSyntaxHighlighter.js (hunk 2 above)
Comment (chenxsan) [reply docs/mdSyntaxHighlighter.js:1]: Hi Nicholas, as there're some discrepancies between eslint docs site and vitepress, e.g., they use shiki for syntax highlight and author their code in TypeScript, while we use Prismjs and JavaScript, hence I have to modify the markdown plugins based on their implementation. So yes, I did copy some code from its repository.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks for cleaning that up!

**Pull request merged**